example/classfile.ts

```diff
@@ -56,33 +56,1255 @@
+        .uint16be("index")
+        .namely("checkcast")
+        .buffer("length", { length: () => 0, formatter: () => 3 }), // checkcast
+      0xc1: Parser.start()
+        .uint16be("index")
+        .namely("instanceof")
+        .buffer("length", { length: () => 0, formatter: () => 3 }), // instanceof
+      0xc2: Parser.start()
+        .namely("monitorenter")
+        .buffer("length", { length: () => 0, formatter: () => 1 }), // monitorenter
+      0xc3: Parser.start()
+        .namely("monitorexit")
+        .buffer("length", { length: () => 0, formatter: () => 1 }), // monitorexit
+      0xc4: "wide", // wide
+      0xc5: Parser.start()
+        .uint16be("index")
+        .uint8("dimensions")
+        .namely("multianewarray")
+        .buffer("length", { length: () => 0, formatter: () => 4 }), // multianewarray
+      0xc6: Parser.start()
+        .int16be("branchoffset")
+        .namely("ifnull")
+        .buffer("length", { length: () => 0, formatter: () => 3 }), // ifnull
+      0xc7: Parser.start()
+        .int16be("branchoffset")
+        .namely("ifnonnull")
+        .buffer("length", { length: () => 0, formatter: () => 3 }), // ifnonnull
+      0xc8: Parser.start()
+        .int32be("branchoffset")
+        .namely("goto_w")
+        .buffer("length", { length: () => 0, formatter: () => 5 }), // goto_w
+      0xc9: Parser.start()
+        .int32be("branchoffset")
+        .namely("jsr_w")
+        .buffer("length", { length: () => 0, formatter: () => 5 }), // jsr_w
+    },
+  });
+
+const BytecodeParser = Parser.start()
+  .useContextVars()
+  .array("instructions", {
+    type: Parser.start().choice("instruction", {
+      tag: function () {
+        switch (
+          this.opcode // Access the "opcode" property of the current item
+        ) {
+          case 0xc4:
+            return 0; // WideInstructionParser
+          case 0xaa:
+            return 1; // TableswitchParser
+          case 0xab:
+            return 2; // LookupswitchParser
+          default:
+            return 3; // InstructionParser (default)
+        }
+      },
+      choices: [
+        // Use an array of choices (indexed numerically)
+        WideInstructionParser,
+        TableswitchParser,
+        LookupswitchParser,
+        InstructionParser,
+      ],
+    }),
+    lengthInBytes: function (item) {
+      return this.$parent.code_length;
+    },
+  });
+
+CodeAttribute = CodeAttribute.useContextVars()
+  .uint16be("max_stack")
+  .uint16be("max_locals")
+  .uint32be("code_length")
+  .nest("code", {
+    type: BytecodeParser,
+  })
+  .uint16be("exception_table_length")
+  .array("exception_table", {
+    type: Parser.start()
+      .uint16be("start_pc")
+      .uint16be("end_pc")
+      .uint16be("handler_pc")
+      .uint16be("catch_type"),
+    length: "exception_table_length",
+  })
+  .uint16be("attributes_count")
+  .array("attributes", {
+    type: "attribute_info",
+    length: "attributes_count",
+  });
```

Comment:
About these two warnings, I used let instead of const, because parser is a bit recursive, and I needed to define the "head" first otherwise it wouldn't let me use it. Something like, parser is not defined yet, maybe you know better.